ir/src/constraints/graph.rs

```diff
@@ -107,14 +107,19 @@ impl AlgebraicGraph {
                 variable_roots,
                 default_domain,
             ),
-            Expression::Rand(index) => {
+            Expression::Rand(name, index) => {
                 // The constraint target for random values defaults to the second (auxiliary) trace
                 // segment.
                 // TODO: make this more general, so random values from further trace segments can be
                 // used. This requires having a way to describe different sets of randomness in
                 // the AirScript syntax.
-                let node_index = self.insert_op(Operation::RandomValue(*index));
-                Ok(ExprDetails::new(node_index, AUX_SEGMENT, default_domain))
+                self.insert_random_access(
+                    symbol_table,
+                    name.name(),
+                    *index,
+                    default_domain,
+                    AUX_SEGMENT,
```

Comment:
nit: Let's keep the order of the domain and segment consistent everywhere. Since we do the segment first when initializing `ExprDetails`, we should do it first here as well